esphome/components/lg_uart/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""
+Simple UART based component to set/poll various settings for LG screens.
+LG provides generic documentation that describes many different things that can be set.
+Not all models/firmware-versions support setting all things.
+
+Additionally, not all models support polling the state of various things.
+Each of the different things that can be controlled can be best represented with a switch (power/mute) or
+    sensor (volume level?) or slider (volume) ... etc
+"""
+
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import uart
+from esphome.const import CONF_UART_ID, CONF_ID
+
+CODEOWNERS = ["@kquinsland"]
+DEPENDENCIES = ["uart"]
+MULTI_CONF = False
+
+# Hub needs the ID of the UART and set ID
+CONF_SCREEN_NUMBER = "screen_number"
+
+# Clients
+CONF_LG_UART_ID = "lg_uart_id"
+CONF_LG_UART_CMD = "lg_uart_cmd"
+
+CONF_DECODE_BASE = "decode_base"
+CONF_DECODE_BASE_TYPES = [10, 16]
+
+
+lg_uart_ns = cg.esphome_ns.namespace("lg_uart")
+LGUartHub = lg_uart_ns.class_("LGUartHub", cg.PollingComponent, uart.UARTDevice)
+
+# the root component
+CONFIG_SCHEMA = cv.COMPONENT_SCHEMA.extend(
+    {
+        cv.GenerateID(): cv.declare_id(LGUartHub),
+        # We need to know which uart device to use
+        cv.Required(CONF_UART_ID): cv.use_id(uart.CONF_ID),
+        # Docs indicate that ID of 0 is broadcast / all screens.
+        # By default, each comes from the factory set as 1.
+        cv.Optional(CONF_SCREEN_NUMBER, default=1): cv.int_range(min=0, max=99),
+    }
+)
+
+# Every component that uses us as a platform will need this
+LG_UART_CLIENT_SCHEMA = cv.Schema(
+    {
+        cv.Required(CONF_LG_UART_ID): cv.use_id(LGUartHub),
```

Comment:
This allows automatic linking if there is only a single `lg_uart` component (The user doesnt have to add the id in yaml)
```suggestion
        cv.GenerateID(CONF_LG_UART_ID): cv.use_id(LGUartHub),
```